src/EditorFeatures/CSharpTest/MakeTypeAbstract/MakeTypeAbstractTests.cs

```diff
@@ -0,0 +1,335 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.MakeTypeAbstract;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.MakeTypeAbstract
+{
+    public class MakeTypeAbstractTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        public MakeTypeAbstractTests(ITestOutputHelper logger)
+          : base(logger)
+        {
+        }
+
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new CSharpMakeTypeAbstractCodeFixProvider());
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsMakeTypeAbstract)]
+        [InlineData("class")]
+        [InlineData("record")]
+        public async Task TestMethod(string typeKind)
```

Comment:
can you revert all of this add only new tests for records.  parameterizing here seems to provide no value and doubles the amount of tests without need.